### PR TITLE
Define iOS module

### PR DESCRIPTION
**Pull request by ened**
This will resolve issues when host apps have `use_frameworks!` removed from their Podfile.

The practice has been taken over from flutter/plugins.

### CODE REVIEW

**Comment by hvisser** [comment]
Thanks for the PR! Is the podspec update something that Flutter upgrades on its own? E.g. is this the same as what a shiny brand new Flutter project would setup?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# next
```
Comment (hvisser) [inline CHANGELOG.md]: Could you move this under 1.1.1? I've already bumped the version, but not released it.

**Comment by ened** [comment]
@hvisser:

```
(base) ➜  tmp flutter create -t plugin module_demo
Creating project module_demo...
  module_demo/LICENSE (created)
  module_demo/ios/.gitignore (created)
  module_demo/ios/Assets/.gitkeep (created)
  module_demo/test/module_demo_test.dart (created)
  module_demo/module_demo.iml (created)
  module_demo/.gitignore (created)
  module_demo/.metadata (created)
  module_demo/android/settings.gradle (created)
  module_demo/android/gradle/wrapper/gradle-wrapper.properties (created)
  module_demo/android/gradle.properties (created)
  module_demo/android/.gitignore (created)
  module_demo/android/src/main/AndroidManifest.xml (created)
  module_demo/android/build.gradle (created)
  module_demo/android/module_demo_android.iml (created)
  module_demo/android/src/main/kotlin/com/example/module_demo/ModuleDemoPlugin.kt (created)
  module_demo/pubspec.yaml (created)
  module_demo/README.md (created)
  module_demo/ios/module_demo.podspec (created)
  module_demo/ios/Classes/ModuleDemoPlugin.m (created)
  module_demo/ios/Classes/SwiftModuleDemoPlugin.swift (created)
  module_demo/ios/Classes/ModuleDemoPlugin.h (created)
  module_demo/lib/module_demo.dart (created)
  module_demo/.idea/runConfigurations/example_lib_main_dart.xml (created)
  module_demo/.idea/libraries/Flutter_for_Android.xml (created)
  module_demo/.idea/libraries/Dart_SDK.xml (created)
  module_demo/.idea/modules.xml (created)
  module_demo/.idea/workspace.xml (created)
  module_demo/CHANGELOG.md (created)
Running "flutter pub get" in module_demo...                         1.7s
  module_demo/example/ios/Runner.xcworkspace/contents.xcworkspacedata (created)
  module_demo/example/ios/Runner.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (created)
  module_demo/example/ios/Runner.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings (created)
  module_demo/example/ios/Runner/Info.plist (created)
  module_demo/example/ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage@2x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage@3x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/LaunchImage.imageset/README.md (created)
  module_demo/example/ios/Runner/Assets.xcassets/LaunchImage.imageset/Contents.json (created)
  module_demo/example/ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-76x76@2x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-29x29@1x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-40x40@1x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-20x20@1x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-83.5x83.5@2x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-20x20@3x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Contents.json (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-20x20@2x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-29x29@3x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-40x40@2x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-60x60@3x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-60x60@2x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-76x76@1x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-40x40@3x.png (created)
  module_demo/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-29x29@2x.png (created)
  module_demo/example/ios/Runner/Base.lproj/LaunchScreen.storyboard (created)
  module_demo/example/ios/Runner/Base.lproj/Main.storyboard (created)
  module_demo/example/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata (created)
  module_demo/example/ios/Runner.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (created)
  module_demo/example/ios/Runner.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings (created)
  module_demo/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme (created)
  module_demo/example/ios/Flutter/Debug.xcconfig (created)
  module_demo/example/ios/Flutter/Release.xcconfig (created)
  module_demo/example/ios/Flutter/AppFrameworkInfo.plist (created)
  module_demo/example/ios/.gitignore (created)
  module_demo/example/test/widget_test.dart (created)
  module_demo/example/module_demo_example.iml (created)
  module_demo/example/.gitignore (created)
  module_demo/example/.metadata (created)
  module_demo/example/android/app/src/profile/AndroidManifest.xml (created)
  module_demo/example/android/app/src/main/res/mipmap-mdpi/ic_launcher.png (created)
  module_demo/example/android/app/src/main/res/mipmap-hdpi/ic_launcher.png (created)
  module_demo/example/android/app/src/main/res/drawable/launch_background.xml (created)
  module_demo/example/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png (created)
  module_demo/example/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png (created)
  module_demo/example/android/app/src/main/res/values/styles.xml (created)
  module_demo/example/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png (created)
  module_demo/example/android/app/src/main/AndroidManifest.xml (created)
  module_demo/example/android/app/src/debug/AndroidManifest.xml (created)
  module_demo/example/android/gradle/wrapper/gradle-wrapper.properties (created)
  module_demo/example/android/gradle.properties (created)
  module_demo/example/android/.gitignore (created)
  module_demo/example/android/settings.gradle (created)
  module_demo/example/android/app/build.gradle (created)
  module_demo/example/android/app/src/main/kotlin/com/example/module_demo_example/MainActivity.kt (created)
  module_demo/example/android/build.gradle (created)
  module_demo/example/android/module_demo_example_android.iml (created)
  module_demo/example/pubspec.yaml (created)
  module_demo/example/README.md (created)
  module_demo/example/ios/Runner/Runner-Bridging-Header.h (created)
  module_demo/example/ios/Runner/AppDelegate.swift (created)
  module_demo/example/ios/Runner.xcodeproj/project.pbxproj (created)
  module_demo/example/lib/main.dart (created)
  module_demo/example/.idea/runConfigurations/main_dart.xml (created)
  module_demo/example/.idea/libraries/Flutter_for_Android.xml (created)
  module_demo/example/.idea/libraries/Dart_SDK.xml (created)
  module_demo/example/.idea/libraries/KotlinJavaRuntime.xml (created)
  module_demo/example/.idea/modules.xml (created)
  module_demo/example/.idea/workspace.xml (created)
Running "flutter pub get" in example...                             1.5s
 
Wrote 100 files.                            

All done!
[✓] Flutter: is fully installed. (Channel stable, v1.17.3, on Mac OS X 10.15.5 19F101, locale en-US)
[✓] Android toolchain - develop for Android devices: is fully installed. (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS: is fully installed. (Xcode 11.5)
[✓] Android Studio: is fully installed. (version 3.6)
[✓] VS Code: is fully installed. (version 1.46.1)
[!] Connected device: is not available.

Run "flutter doctor" for information about installing additional components.

In order to run your application, type:

  $ cd module_demo/example
  $ flutter run

Your application code is in module_demo/example/lib/main.dart.

Your plugin code is in module_demo/lib/module_demo.dart.

Host platform code is in the "android" and "ios" directories under module_demo.
To edit platform code in an IDE see https://flutter.dev/developing-packages/#edit-plugin-package.

(base) ➜  tmp tail module_demo/ios/module_demo.podspec 
  s.author           = { 'Your Company' => 'email@example.com' }
  s.source           = { :path => '.' }
  s.source_files = 'Classes/**/*'
  s.dependency 'Flutter'
  s.platform = :ios, '8.0'

  # Flutter.framework does not contain a i386 slice. Only x86_64 simulators are supported.
  s.pod_target_xcconfig = { 'DEFINES_MODULE' => 'YES', 'VALID_ARCHS[sdk=iphonesimulator*]' => 'x86_64' }
  s.swift_version = '5.0'
end
(base) ➜  tmp flutter --version
Flutter 1.17.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b041144f83 (3 weeks ago) • 2020-06-04 09:26:11 -0700
Engine • revision ee76268252
Tools • Dart 2.8.4
```


Will update the CHANGELOG entry next.

**Comment by hvisser** [comment]
I guess that was a yes 😅  Think we're good to merge this.

**Pull request merged**